js/index.js

```diff
@@ -0,0 +1,265 @@
+const apiKey = '570c92340d384eea8eb180128200806';
+const apiURL = 'https://api.weatherapi.com/v1/forecast.json';
+const corsHack = 'https://cors-anywhere.herokuapp.com/';
+const apiImageURL =
+  'https://cors-anywhere.herokuapp.com/https://www.flickr.com/services/rest/?method=flickr.photos.search&tags=nature,spring,morning&tag_mode=all&extras=url_h&format=json&nojsoncallback=1';
+const apiKeyImage = '8cb61d73c757a9c87c9ca27d5ce9b80b';
+const apiKeyGeo = 'dcb27ce8a7e240bc81680f4b36b918a9';
+const apiGeolocationUrl =
+  'https://api.opencagedata.com/geocode/v1/json?q=Minsk&pretty=1&no_annotations=1';
+
+const getCityWeather = (city) =>
+  fetch(
+    `${corsHack}${apiURL}?key=${apiKey}&q=${city}&lang=en&days=3`,
+  ).then((r) => r.json());
+
+const getImageBackground = () => {
+  return fetch(`${corsHack}${apiImageURL}&api_key=${apiKeyImage}`).then((r) =>
+    r.json(),
+  );
+};
+
+const getGeoLocation = () => {
+  return fetch(`${corsHack}${apiGeolocationUrl}&key=${apiKeyGeo}`).then((r) =>
+    r.json(),
+  );
+};
+
+const backgroundUpdate = async () => {
+  const response = await getImageBackground();
+
+  function* imagesGenerator() {
+    for (let photo of response.photos.photo) {
+      yield photo.url_h;
+    }
+  }
+  const images = imagesGenerator();
+  images.next(); // undefined
+  document.body.style.background = `url(${images.next().value})`;
+
+  console.log(images.next().value);
+
+  setInterval(() => {
+    const curImage = images.next().value;
+    document.body.style.background = `url(${curImage || images.next().value})`;
+  }, 40000);
+};
+backgroundUpdate();
+
+class Map {
```

Comment:
Let's update the name of this class to smth. else, for example `LocationMap` or `MapData`. As a rule - you should not use js reserved words or existing embedded js names of classes in your code for names of your classes, properties, methods etc (for ex. `Map, Set, Array, new, class, catch, throw, const, string, etc.`).